src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1630,6 +1654,61 @@ private void MakeSeparatorList(ReadOnlySpan<char> separators, ref ValueListBuild
             }
         }
 
+        private void MakeSeparatorListVectorized(ref ValueListBuilder<int> sepListBuilder, char c, char? c2 = null, char? c3 = null)
+        {
+            Vector256<byte> shuffleConstant = Vector256.Create(0x02, 0x06, 0x0A, 0x0E, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
+                                                        0xFF, 0xFF, 0xFF, 0xFF, 0x02, 0x06, 0x0A, 0x0E, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF);
+
+            Vector256<ushort> v1 = Vector256.Create(c);
+            Vector256<ushort>? v2 = c2 is char sep2 ? Vector256.Create(sep2) : (Vector256<ushort>?)null;
+            Vector256<ushort>? v3 = c3 is char sep3 ? Vector256.Create(sep3) : (Vector256<ushort>?)null;
+
+            ref char c0 = ref MemoryMarshal.GetReference(this.AsSpan());
+            int cond = Length - (Length % Vector256<ushort>.Count);
+            int i = 0;
+
+            for (; i < cond; i += Vector256<ushort>.Count)
+            {
+                ref char ci = ref Unsafe.Add(ref c0, i);
+                Vector256<ushort> charVector = Unsafe.As<char, Vector256<ushort>>(ref ci);
+                Vector256<ushort> cmp = Avx2.CompareEqual(charVector, v1);
+
+                if (v2 is Vector256<ushort> vecSep2)
+                {
+                    cmp = Avx2.Or(Avx2.CompareEqual(charVector, vecSep2), cmp);
+                }
+
+                if (v3 is Vector256<ushort> vecSep3)
+                {
+                    cmp = Avx2.Or(Avx2.CompareEqual(charVector, vecSep3), cmp);
+                }
+
+                if (Avx.TestZ(cmp, cmp)) { continue; }
+
+                Vector256<byte> mask = Avx2.ShiftLeftLogical(cmp.AsUInt64(), 4).AsByte();
+                mask = Avx2.Shuffle(mask, shuffleConstant);
+
+                Vector128<byte> res = Sse2.Or(Avx2.ExtractVector128(mask, 0), Avx2.ExtractVector128(mask, 1));
```

Comment:
```suggestion
                Vector128<byte> res = Sse2.Or(mask.GetLower(), mask.GetUpper());
```
results AFAIK in better codegen, as for "lower" the xmm-part of the ymm-register is just reused instead of emitting a `vextracti128`-instruction.

Reply:
Typed this into [Sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgDkBXfGKASzFwBuGvSasAShwB2GXtxYBJGXym5+QkY2YtJMuTEXLeq9SwAaADiTDaAZiakGAYQYBvGg09N7ANRhgMaHJSCwAeYABPDBgAPgYAWQAKPwDoUgBWJHCo2IYANwBKNw8vUuIAdgYAQTyEUhYAeShEvJYAcRgMAFUABx6eRIK0fPbOgBkIAHcBgoKbUoBfEs9iX39AqGCwyOi4+NJk9bTM7N38ovdqUrLKmrrG5rv6gFEEDChsAJSNrZbh8iG1VqLzeHy+R02IT+DDos3mXiW1AWQA). Seems like it is indeed better codegen!